### DISCUSSION TITLE
[google-apps-script] Remove namespace for GoogleAppsScript.Base.MimeType

### DISCUSSION THREAD

**Discussion by KaiShoya** [DefinitelyTyped/DefinitelyTyped#61917] [Issues with a @types package] [answered]
> An enumeration that provides access to MIME-type declarations without typing the strings explicitly. Methods that expect a MIME type rendered as a string (for example, 'image/png') also accept any of the values below, so long as the method supports the underlying MIME type.

The [GoogleAppsScript.Base.MimeType Doc](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e48ed28b659609b706e9ff77cba7eece8da57be4/types/google-apps-script/google-apps-script.base.d.ts#L176) says this, but there are some problems.

This code works on GAS, but I get an error "Cannot find name 'MimeType'. ts(2304)".
```js
const file = DriveApp.getFileById('xxxxxxxxxxxxxxxx')
// check MimeType
if (file.getMimeType() === MimeType.GOOGLE_SHEETS) {
  // some processing
}
```
<img width="503" alt="Screen Shot 2022-08-24 at 20 35 05" src="https://user-images.githubusercontent.com/90382293/186408496-fd79db99-02bf-4c44-9ed9-6e019d452cc3.png">

> If you want to use MimeType.GOOGLE_SLIDES, you can write GoogleAppsScript.Base.MimeType.GOOGLE_SLIDES.

If I fix it according to [this comment](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/36546#issuecomment-547475151), it will not work on GAS.(I'm using [clasp](https://github.com/google/clasp) to deploy.)
Also, I get an another error, but it's explained in [another discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/61140), so I won't go into it here.
```ts
const file = DriveApp.getFileById('xxxxxxxxxxxxxxxx')
// check MimeType
if (file.getMimeType() === GoogleAppsScript.Base.MimeType.GOOGLE_SHEETS) {
  // some processing
}
```
<img width="663" alt="Screen Shot 2022-08-24 at 21 49 01" src="https://user-images.githubusercontent.com/5924971/186422470-b90b677f-3a80-4e81-91dd-b18e60a4eee8.png">

I think it would be better to remove the GoogleAppsScript.Base namespace so that MimeType can be used.

Thank you:)

[a comment by an automated account is not shown]

**Comment by mtgto** [comment, marked as answer]
Thanks @KaiShoya .

> If I fix it according to https://github.com/DefinitelyTyped/DefinitelyTyped/issues/36546#issuecomment-547475151, it will not work on GAS.(I'm using [clasp](https://github.com/google/clasp) to deploy.)

Yes, you're right. GAS not provides global variable `GoogleAppsScript`.
[My comment](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/36546#issuecomment-547475151) is wrong.
I find GAS runs this code fine.

```js
Logger.log(MimeType.GOOGLE_APPS_SCRIPT); // "application/vnd.google-apps.script"
```

However, you already write, this code can not compile in tsc.
The reason is `MimeType` is not declared as global variable.
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/50858e11bf68ef9c81bd8f6c7bfe6327d0dc05e7#diff-b6f71578b72389f6f6e239398e685f7e6ec4c8b004d7f9503640a984b5bf1908R277

To fix it, I create pull request.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61918
